trieste/objectives/multi_objectives.py

```diff
@@ -236,3 +237,76 @@ def gen_pareto_optimal_points(n: int, seed: int | None = None) -> TensorType:
         search_space=Box([0.0], [1.0]) ** d,
         gen_pareto_optimal_points=gen_pareto_optimal_points,
     )
+
+
+@dataclass(frozen=True)
+class ConstrainedMultiObjectiveTestProblem(MultiObjectiveTestProblem):
+    """
+    Convenience container class for synthetic constrained multi-objective test functions, containing
+    a generator for the pareto optimal points, which can be used as a reference of performance
+    measure of certain multi-objective optimization algorithms.
+    """
+
+    constraint: Callable[[TensorType], TensorType]
+    """The synthetic test function's constraints"""
+
+
+def ConstrainedBraninCurrin() -> ConstrainedMultiObjectiveTestProblem:
+    """
+    The ConstrainedBraninCurrin problem, typically evaluated over :math:`[0, 1]^2`.
+    See :cite:`belakaria2019max` and :cite:`daulton2020differentiable`
+    (the latter for adding the constraint) for details.
+
+    :return: The problem specification.
+    """
+
+    def gen_pareto_optimal_points(n: int, seed: int | None = None) -> TensorType:
+        return tf.zeros(shape=0)
+
+    def evaluate_slack_true(x: TensorType) -> TensorType:
```

Comment:
I'm not familiar with the constrained brannin currin function, would it work with any other threshold than 0?

Reply:
but in any case, in general threshold will typically be flexible and user could modify it - so in the new method in the interface lets have an additional argument `threshold: float`, you can set it to 0 as a default in your problem and potentially add to the docs that it shouldn't be used in any other way if it only works with 0

Reply:
>would it work with any other threshold than 0?  

the constraint should be well defined on the different thresholds. The default formulation used here is just a copy of the reference paper's equation since its property is better understood (than other thresholds).

Reply:
ok, then lets simply set it as a default and say something about it in docs